StartioAdmobMediation/StartioAdmobCustomEvent.m

```diff
@@ -0,0 +1,146 @@
+#import "StartioAdmobRewardedAdapter.h"
+#import "StartioAdmobBannerAdapter.h"
+#import "StartioAdmobNativeAdapter.h"
+#import "StartioAdmobParameters.h"
+
+@import StartApp;
+
+static NSString* const kSettingsParameterKey = @"parameter";
+
+@interface StartioAdmobCustomEvent ()
+@property (nonatomic, strong) StartioAdmobInterstitialAdapter *interstitialAdapter;
+@property (nonatomic, strong) StartioAdmobRewardedAdapter *rewardedAdapter;
+@property (nonatomic, strong) StartioAdmobBannerAdapter *bannerAdapter;
+@property (nonatomic, strong) StartioAdmobNativeAdapter *nativeAdapter;
+@end
+
+@implementation StartioAdmobCustomEvent
+
++ (GADVersionNumber)adapterVersion {
+    return [self GADVersionFromString:StartioAdmobAdapterVersion];
+}
+
++ (GADVersionNumber)adSDKVersion {
+    __block NSString *version = nil;
+    if ([NSThread isMainThread]) {
+        version = [[STAStartAppSDK sharedInstance] version];
+    }
+    else {
+        dispatch_sync(dispatch_get_main_queue(), ^{
+            version = [[STAStartAppSDK sharedInstance] version];
+        });
+    }
+    return [self GADVersionFromString:version];
+}
+
++ (nullable Class<GADAdNetworkExtras>)networkExtrasClass {
+    return nil;
+}
+
++ (void)setUpWithConfiguration:(GADMediationServerConfiguration *)serverConfiguration completionHandler:(GADMediationAdapterSetUpCompletionBlock)completionHandler {
+    if (serverConfiguration.credentials.count == 0) {
+        completionHandler([NSError errorWithDomain:GADErrorDomain code:GADErrorMediationAdapterError userInfo:@{NSLocalizedDescriptionKey : @"No mdiation parameters received"}]);
+    }
+    else {
+        StartioAdmobParameters *mediationParameters = [[StartioAdmobParameters alloc] initWithParametersJSON:serverConfiguration.credentials[0].settings[@"parameter"]];
+        if (mediationParameters.appId.length != 0) {
+            if ([NSThread isMainThread]) {
+                [self setupStartioSDKWithMediationParameters:mediationParameters testAdsEnabled:NO];
+                completionHandler(nil);
+            }
+            else {
+                dispatch_sync(dispatch_get_main_queue(), ^{
+                    [self setupStartioSDKWithMediationParameters:mediationParameters testAdsEnabled:NO];
+                });
+                completionHandler(nil);
+            }
+        }
+        else {
+            completionHandler([NSError errorWithDomain:GADErrorDomain code:GADErrorMediationAdapterError userInfo:@{NSLocalizedDescriptionKey : @"No Start.io AppID provided yet"}]);
+        }
+    }
+}
+
++ (void)setupStartioSDKWithMediationParameters:(StartioAdmobParameters *)mediationParameters testAdsEnabled:(BOOL)testAdsEnabled {
+    STAStartAppSDK *sdk = [STAStartAppSDK sharedInstance];
+    
+    sdk.appID = mediationParameters.appId;
+    sdk.returnAdEnabled = NO;
+    sdk.consentDialogEnabled = NO;
+    [sdk addWrapperWithName:@"AdMob" version:StartioAdmobAdapterVersion];
+    sdk.testAdsEnabled = YES;
```

Comment:
here sdk.testAdsEnabled = YES; is duplicated
is it needed to  overwrite this for debug?